src/core/Akka.Remote/Endpoint.cs

```diff
@@ -411,147 +414,174 @@ protected override void PostRestart(Exception reason)
             throw new IllegalActorStateException("BUG: ReliableDeliverySupervisor has been attempted to be restarted. This must not happen.");
         }
 
-        protected override void OnReceive(object message)
+        protected void Receiving()
         {
-            message.Match()
-                .With<EndpointWriter.FlushAndStop>(flush =>
-                {
-                    //Trying to serve untilour last breath
-                    ResendAll();
-                    _writer.Tell(EndpointWriter.FlushAndStop.Instance);
-                    Context.Become(FlushWait);
-                })
-                .With<EndpointManager.Send>(HandleSend)
-                .With<Ack>(ack =>
+            Receive<EndpointWriter.FlushAndStop>(flush =>
+            {
+                //Trying to serve until our last breath
+                ResendAll();
+                _writer.Tell(EndpointWriter.FlushAndStop.Instance);
+                Become(FlushWait);
+            });
+            Receive<IsIdle>(idle => { }); // Do not reply, we will Terminate soon, or send a GotUid
```

Comment:
New message class that needs to be handled for checking if the `ReliableDeliverySupervisor` is an idle state or not.